src/net_processing.cpp

```diff
@@ -321,18 +321,12 @@ struct CNodeState {
     bool fPreferHeaderAndIDs;
     /**
       * Whether this peer will send us cmpctblocks if we request them.
-      * This is not used to gate request logic, as we really only care about fSupportsDesiredCmpctVersion,
-      * but is used as a flag to "lock in" the version of compact blocks (fWantsCmpctWitness) we send.
       */
     bool fProvidesHeaderAndIDs;
```

Comment:
Yes, these parameters are currently confusing (hence this PR to clean them up!)

`fProvidersHeaderAndIDs` is currently used to make sure that the compact blocks version doesn't change after negotiation. If a peer sends `sendcmpct(version=2)`, then any future `sendcmpct` messages where version!=2 have no effect. The peer can change hb mode by sending `sendcmpct(version=2, hb_mode={true|false})`.

This is going away with this PR because the software will only recognize version 2 sendcmpct messages from now on.